texar/modules/classifiers/xlnet_classifier.py

```diff
@@ -0,0 +1,173 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+XLNet classifiers.
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import tensorflow as tf
+from texar import global_mode
+from texar.core import layers
+from texar.modules.classifiers.classifier_base import ClassifierBase
+from texar.modules import XLNetEncoder
+from texar.utils import utils
+from texar.hyperparams import HParams
+
+# pylint: disable=too-many-arguments, invalid-name, no-member,
+# pylint: disable=too-many-branches, too-many-locals, too-many-statements
+
+__all__ = [
+    "XLNetClassifier"
+]
+
+
+class XLNetClassifier(ClassifierBase):
+    """Classifier based on XLNet modules.
+
+    This is a combination of the :class:`~texar.modules.XLNetEncoder` with a
+    classification layer. Both step-wise classification and sequence-level
+    classification are supported, specified in :attr:`hparams`.
+
+    Arguments are the same as in :class:`~texar.modules.XLNetEncoder`.
+
+    Args:
+        pretrained_model_name (optional): a str with the name
+            of a pre-trained model to load. Currently only 'xlnet-large-cased'
+            is supported. If `None`, will use the model name in :attr:`hparams`.
+        cache_dir (optional): the path to a folder in which the
+            pre-trained models will be cached. If `None` (default),
+            a default directory will be used.
+        hparams (dict or HParams, optional): Hyperparameters. Missing
+            hyperparameter will be set to default values. See
+            :meth:`default_hparams` for the hyperparameter structure
+            and default values.
+
+    .. document private functions
+    .. automethod:: _build
+    """
+
+    def __init__(self,
+                 pretrained_model_name=None,
+                 cache_dir=None,
+                 hparams=None):
+        ClassifierBase.__init__(self, hparams)
+
+        with tf.variable_scope(self.variable_scope):
+            # Creates the underlying encoder
+            encoder_hparams = utils.dict_fetch(
+                hparams, XLNetEncoder.default_hparams())
+            if encoder_hparams is not None:
+                encoder_hparams['name'] = None
+            self._encoder = XLNetEncoder(
+                pretrained_model_name=pretrained_model_name,
+                cache_dir=cache_dir,
+                hparams=encoder_hparams)
+            if self._hparams.use_projection:
+                self.projection = layers.get_layer(hparams={
+                    "type": "Dense",
+                    "kwargs": {
+                        "units": self._encoder.output_size
+                    }
+                })
+
+            # summary type
+            if self._hparams.summary_type == 'last':
```

Comment:
This part need to be updated (see the pytorch version) -- `summary_type` --> `cls_strategy`

Reply:
Done! [9280366]